File: number_theory/01_primality_test.cpp
#include<iostream>
using namespace std;

bool is_prime(int n){
	if( n <= 1 )
		return false;
	bool res = true;
	for( int i = 2; i*i <= n; i++ ){
		if( n % i == 0 ){
			res = false;
			break;
		}
	}
	return res;
}

int main(){
	int n;
	cout << "Enter number : ";
	cin >> n;
	cout << ((is_prime(n) ? "prime" : "not prime")) << endl;
	return 0;
}

File: number_theory/01_primality_test.py


def is_prime(n : int) -> bool:
	if( n <= 1 ):
		return False
	res = True
	i = 2
	while( i*i <= n ):
		if( n % i == 0 ):
			res = False
			break
		i += 1
	return res

n : int = int(input("Enter number : "))
print( "prime" if is_prime(n) else "not prime")